### PR TITLE
Migrate deployment mechanism to gcloud app deploy

### PR DESCRIPTION
**Pull request by fejta**
/assign @rmmh 

I deployed with `make deploy VERSION=test`

https://test-dot-k8s-gubernator.appspot.com/pr

### CODE REVIEW

File: gubernator/Makefile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	@echo TODO(kubernetes/test-infra/3884): deploy without needing this file
+	@echo Ask Google GKE EngProd how to create this file.
+
+secrets.json:
+	@echo TODO(kubernetes/test-infra/3884): deploy without needing this file
+	@echo Ask Google GKE EngProd how to create this file.
+
+deploy: github/webhook_secret secrets.json
+	gcloud app deploy --version="$(VERSION)" --project="$(PROJECT)" --no-promote --quiet
+	cd github && gcloud app deploy --version="$(VERSION)" --project="$(PROJECT)" --no-promote --quiet
```
Comment (rmmh) [inline gubernator/Makefile]: echo "Deployed staging to https://$(VERSION)-dot-k8s-gubernator.appspot.com"

File: gubernator/app.yaml [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
 runtime: python27
 api_version: 1
 threadsafe: yes
 instance_class: F2
 
+inbound_services:
+- warmup
+
 handlers:
+- url: /_ah/warmup
```
Comment (rmmh) [inline gubernator/app.yaml]: Why not warmup with / ?

File: gubernator/Makefile [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+# TOOD(fejta): consider the following default:
+# gcloud app versions list --sort-by=~LAST_DEPLOYED --limit=1 --format='value(id)'
+migrate: SHELL:=/bin/bash
+migrate: versions
+	@read -p "Promote $(SERVICE) [$(VERSION)]: " version && \
+	gcloud app versions migrate "$${version:-$(VERSION)}" --quiet && \
+	gcloud app versions migrate "$${version:-$(VERSION)}" --service=github --quiet
+
+
+delete: SHELL:=/bin/bash
```
Comment (rmmh) [inline gubernator/Makefile]: if we're deploying each version to a git-tagged version, we need some way to delete old (possibly insecure) deployments

Comment (fejta) [inline gubernator/Makefile]: That is what this does -- lists the deployed versions and asks which one you want to delete

File: gubernator/app.yaml (hunk 2 above)
Comment (fejta) [inline gubernator/app.yaml]: I'm just following the warmup example docs: https://cloud.google.com/appengine/docs/standard/python/warmup-requests/configuring

I suspect that `/_ah/warmup` is the URL that appengine calls during migration to warmup your app, and it is up to the app ensure that this calls does what is necessary for warmup.

I expect future PRs to expand this handler to do something more appropriate, but this handler is necessary to make `gcloud app versions migrate` work at all.

File: gubernator/Makefile (hunk 1 above)
Comment (fejta) [inline gubernator/Makefile]: `gcloud app deploy` outputs this information

**Pull request merged**